elide-async/src/main/java/com/yahoo/elide/async/hooks/TableExportHook.java

```diff
@@ -7,44 +7,61 @@
+import com.yahoo.elide.async.operation.GraphQLTableExportCallableOperation;
 import com.yahoo.elide.async.service.AsyncExecutorService;
-import com.yahoo.elide.async.service.thread.TableExportCallable;
+import com.yahoo.elide.core.exceptions.InvalidOperationException;
 import com.yahoo.elide.core.security.ChangeSpec;
 import com.yahoo.elide.core.security.RequestScope;
 
+import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.Callable;
 
 /**
  * LifeCycle Hook for execution of TableExpoer.
  */
 public class TableExportHook extends AsyncAPIHook<TableExport> {
+    Map<ResultType, TableExportFormatter> supportedFormatters;
 
-    public TableExportHook (AsyncExecutorService asyncExecutorService, Integer maxAsyncAfterSeconds) {
+    public TableExportHook (AsyncExecutorService asyncExecutorService, Integer maxAsyncAfterSeconds,
+            Map<ResultType, TableExportFormatter> supportedFormatters) {
         super(asyncExecutorService, maxAsyncAfterSeconds);
+        this.supportedFormatters = supportedFormatters;
     }
 
     @Override
-    public void execute(Operation operation, TransactionPhase phase, TableExport query, RequestScope requestScope,
+    public void execute(Operation operation, TransactionPhase phase, TableExport export, RequestScope requestScope,
             Optional<ChangeSpec> changes) {
-        AsyncExecutorService service = getAsyncExecutorService();
-        TableExporter exporter = new TableExporter(service.getElide(), requestScope.getApiVersion(),
-                requestScope.getUser());
-        TableExportCallable queryWorker = new TableExportCallable(query, service.getResultStorageEngine(), exporter);
-        executeHook(operation, phase, query, requestScope, queryWorker);
+        Callable<AsyncAPIResult> callable = getOperation(export, requestScope);
+        executeHook(operation, phase, export, requestScope, callable);
     }
 
     @Override
-    public void validateOptions(AsyncAPI query, RequestScope requestScope) {
-        super.validateOptions(query, requestScope);
+    public void validateOptions(AsyncAPI export, RequestScope requestScope) {
+        super.validateOptions(export, requestScope);
     }
 
     @Override
-    public Callable getOperation(AsyncAPI query, RequestScope requestScope) {
-        // TODO Auto-generated method stub
-        return null;
+    public Callable<AsyncAPIResult> getOperation(AsyncAPI export, RequestScope requestScope) {
+        Callable operation = null;
+        TableExportFormatter formatter = supportedFormatters.get(((TableExport) export).getResultType());
+
+        if (formatter == null) {
+            throw new InvalidOperationException("Formatter unavailable for " + ((TableExport) export).getResultType());
+        }
+
+        if (export.getQueryType().equals(QueryType.GRAPHQL_V1_0)) {
+            operation = new GraphQLTableExportCallableOperation(formatter, getAsyncExecutorService(), export,
```

Comment:
This is a really long noun.  I would remove either Callable or Operation from all of these classes.